packages/hardhat/test/integration/workflows/proxy_payment_addresses.js

```diff
@@ -0,0 +1,229 @@
+/** 
+  Projects can deploy addresses that will forward funds received to the project's funding cycle.
+*/
+
+const { BigNumber } = require("ethers");
+
+// The currency will be 0, which corresponds to ETH, preventing the need for currency price conversion.
+const currency = 0;
+
+module.exports = [
+  {
+    description: "Deploy a project",
+    fn: async ({
+      constants,
+      contracts,
+      executeFn,
+      randomBigNumberFn,
+      BigNumber,
+      randomBytesFn,
+      randomStringFn,
+      randomSignerFn,
+      incrementFundingCycleIdFn,
+      incrementProjectIdFn,
+    }) => {
+      const expectedProjectId = incrementProjectIdFn();
+
+      // Burn the unused funding cycle ID id.
+      incrementFundingCycleIdFn();
+
+      const owner = randomSignerFn();
+
+      // Make the test case cleaner with a reserved rate of 0.
+      const reservedRate = BigNumber.from(0);
+
+      await executeFn({
+        caller: randomSignerFn(),
+        contract: contracts.terminalV1,
+        fn: "deploy",
+        args: [
+          owner.address,
+          randomBytesFn({
+            // Make sure its unique by prepending the id.
+            prepend: expectedProjectId.toString(),
+          }),
+          randomStringFn(),
+          {
+            target: BigNumber.from(1),
+            currency,
+            duration: randomBigNumberFn({
+              min: BigNumber.from(0),
+              max: constants.MaxUint16,
+            }),
+            cycleLimit: randomBigNumberFn({
+              max: constants.MaxCycleLimit,
+            }),
+            discountRate: randomBigNumberFn({ max: constants.MaxPercent }),
+            ballot: constants.AddressZero,
+          },
+          {
+            reservedRate,
+            bondingCurveRate: randomBigNumberFn({
+              max: constants.MaxPercent,
+            }),
+            reconfigurationBondingCurveRate: randomBigNumberFn({
+              max: constants.MaxPercent,
+            }),
+          },
+          [],
+          [],
+        ],
+      });
+      return { owner, reservedRate, expectedProjectId };
+    },
+  },
+  {
+    description:
+      "Make sure the terminalV1 got set as the project's current terminal",
+    fn: ({
+      checkFn,
+      contracts,
+      randomSignerFn,
+      local: { expectedProjectId },
+    }) =>
+      checkFn({
+        caller: randomSignerFn(),
+        contract: contracts.terminalDirectory,
+        fn: "terminalOf",
+        args: [expectedProjectId],
+        expect: contracts.terminalV1.address,
+      }),
+  },
+  {
+    description: "Deploy a proxy payment address",
+    fn: ({
+      executeFn,
+      deployer,
+      contracts,
+      randomStringFn,
+      local: { expectedProjectId },
+    }) =>
+      executeFn({
+        caller: deployer,
+        contract: contracts.proxyPaymentAddressManager,
+        fn: "deploy",
+        args: [expectedProjectId, randomStringFn()],
+      }),
+  },
+  {
```

Comment:
add a block to verify that the address got saved in the addressesOf array

Reply:
This is actually already done.